### PR TITLE
Adding routes and components for each button for Courses

### PR DESCRIPTION
**Pull request by yaoxing1990**
Implemented course button functions and fixed coding styles

### CODE REVIEW

File: src/app/courses/courses.module.ts [hunk 1]
```diff
@@ -23,7 +28,13 @@ import { MaterialModule } from '../shared/material.module';
...
-    CoursesRequestComponent
-  ]
+    CoursesRequestComponent,
+    CourseOpenComponent,
+    CourseManageComponent,
+    CourseViewComponent,
+    CourseCreditsComponent,
+    CourseProgressComponent
+  ],
+  providers: [ CourseValidatorService ]
```
Comment (razu9861) [inline src/app/courses/courses.module.ts]: Course validator is now changed to ValidatorService please change it, and do import validator service

File: src/app/courses/open-course/course-open.component.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<h1>Open Course {{id}}</h1>
```
Comment (nirojdyola) [inline src/app/courses/open-course/course-open.component.html]: Do we need to display id or courseTitle ? Could you please confirm it ?

File: src/app/courses/progress-course/course-progress.component.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<h1>Course Progress of {{id}}</h1>
```
Comment (nirojdyola) [inline src/app/courses/progress-course/course-progress.component.html]: Do we need to display id or courseTitle ? Could you please confirm it ?

File: src/app/courses/view-course/course-view.component.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<h1>View Course {{id}}</h1>
```
Comment (nirojdyola) [inline src/app/courses/view-course/course-view.component.html]: Do we need to display id or courseTitle ? Could you please confirm it ?

**Comment by yaoxing1990** [comment]
@nirojdyola Updated as suggested. Added couchdb service in each button, so course content can be retrieved by id.

File: src/app/courses/credits-course/course-credits.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<h1>Course credits of {{courses}}</h1>
```
Comment (razu9861) [inline src/app/courses/credits-course/course-credits.component.html]: {{courses}} is it for all courses or individual course,
Please follow style guide for naming convention

File: src/app/courses/courses.module.ts [hunk 6]
```diff
@@ -23,7 +29,13 @@ import { MaterialModule } from '../shared/material.module';
...
-    CoursesRequestComponent
-  ]
+    CoursesRequestComponent,
+    CourseOpenComponent,
+    CourseManageComponent,
+    CourseViewComponent,
+    CourseCreditsComponent,
+    CourseProgressComponent
+  ],
+  providers: [ ValidatorService ]
```
Comment (paulbert) [inline src/app/courses/courses.module.ts]: Please remove this & the import statement from this file.  The `ValidatorService` is included in `app.module.ts` and does not need to be included as a provider in any of the other modules.

File: src/app/courses/credits-course/course-credits.component.html (hunk 5 above)
Comment (yaoxing1990) [reply src/app/courses/credits-course/course-credits.component.html]: @razu9861 Updated. See https://github.com/ole-vi/planet/pull/259/commits/9d0e4a478ece1dfc7ee0ff4881b637947ded8b37

File: src/app/courses/courses.module.ts (hunk 6 above)
Comment (yaoxing1990) [reply src/app/courses/courses.module.ts]: @paulbert https://github.com/ole-vi/planet/pull/259/commits/c95c8c73d1d139a251dc9f0f7d506caba2328e5a

**Comment by yaoxing1990** [comment]
@paulbert @razu9861 @nirojdyola @lmmrssa @Pr0chin 
Implemented "Manage Course" button. Several problems still exists:
* "Course Level" dropdown menu is not available. Same problem exists in "Add course" button as well.
* "Frequency" is not populated
* Update data to couch db is not successful

**Comment by lmmrssa** [comment]
Error Received
```
 92% chunk asset optimization                                                 y <--- Last few GCs --->

  497502 ms: Mark-sweep 650.1 (1167.4) -> 449.3 (1143.4) MB, 3961.7 / 0.0 ms [allocation failure] [GC in old space requested].
  499260 ms: Mark-sweep 449.3 (1399.4) -> 449.1 (1343.4) MB, 1756.6 / 0.0 ms [allocation failure] [GC in old space requested].
  500839 ms: Mark-sweep 449.1 (1599.4) -> 446.9 (1450.4) MB, 1577.0 / 0.0 ms [last resort gc].
  502102 ms: Mark-sweep 446.9 (1450.4) -> 446.9 (1419.4) MB, 1262.9 / 0.0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x2e04a07cf781 <JS Object>
    1: flatten_args(aka flatten_args) [0x2e04a0704381 <undefined>:~11681] [pc=0xc7593c3eb59] (this=0x2e04a0704381 <undefined>,fn=0x358135d93731 <an AST_Function with map 0x112ae206ce79>,scope=0x358135d95499 <an AST_Function with map 0x196f7e2b9a59>)
    2: /* anonymous */(aka /* anonymous */) [0x2e04a0704381 <undefined>:11627] [pc=0xc7593dd885f] (this=0x2e04a0704381 <undefined>,self=0x2e2f9f6e0...

FATAL ERROR: Committing semi space failed. Allocation failed - process out of memory
 1: node::Abort() [ng]
 2: 0x109f13c [ng]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [ng]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [ng]
 5: v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [ng]
 6: v8::internal::Heap::CollectGarbage(v8::internal::GarbageCollector, char const*, char const*, v8::GCCallbackFlags) [ng]
 7: v8::internal::Factory::NewRawOneByteString(int, v8::internal::PretenureFlag) [ng]
 8: v8::internal::Factory::NewStringFromOneByte(v8::internal::Vector<unsigned char const>, v8::internal::PretenureFlag) [ng]
 9: v8::internal::Factory::NumberToString(v8::internal::Handle<v8::internal::Object>, bool) [ng]
10: v8::internal::Object::ToString(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>) [ng]
11: v8::internal::Object::ConvertToName(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>) [ng]
12: v8::internal::LookupIterator::PropertyOrElement(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, bool*, v8::internal::LookupIterator::Configuration) [ng]
13: v8::internal::Runtime::GetObjectProperty(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>) [ng]
14: 0xed4961 [ng]
15: v8::internal::Runtime_KeyedGetProperty(int, v8::internal::Object**, v8::internal::Isolate*) [ng]
16: 0xc758e8092a7
Aborted
```

**Comment by lmmrssa** [comment]
@yaoxing1990 your rebase went wrong. Its showing everything from master as change.

**Comment by paulbert** [comment]
The change log is very bizarre for this, so I'm going to close it.

**Pull request closed without merging**